### ISSUE TITLE
[REQ] [swift6] request interceptor needs RequestBuilder

### ISSUE THREAD

**Issue by Jonas1893** [OpenAPITools/openapi-generator#19846]
### Is your feature request related to a problem? Please describe.

I really like the introduction of request interceptors into the new swift6 URLSession template but I think at one point it falls short:
It is not possible to use the interceptor to decide whether the request `requiresAuthentication` and thus needs a bearer token or not. 

This was possible before using the swift5 generator with Alamofire template: There `RequestInterceptors` could be injected in the `RequestBuilder` for every request like this
```
override func createAlamofireSession(interceptor: RequestInterceptor? = nil) -> Alamofire.Session {

        return super.createAlamofireSession(
            interceptor: requiresAuthentication ? BearerTokenHandler() : UnauthenticatedRequestInterceptor()
        )
    }
``` 
and we could easily use `RequestBuilder`s `requiresAuthentication` to decide whether we need to pass the bearer token or not. 

Now there is only a global `RequestInterceptor` available on the `OpenAPIClient` type. This `RequestInterceptor` however has no relation to `RequestBuilder` and therefor we do not know whether this request `requiresAuthentication` or not. 

## Describe the solution you'd like

Pass a reference of `RequestBuilder` to `RequestInterceptor`, method signature would then look like this:
```
func intercept(urlRequest: URLRequest, urlSession: URLSessionProtocol, requestBuilder: `RequestBuilder`, completion: @escaping (Result<URLRequest, Error>) -> Void)
```

Since it is possible to access `openAPIClient` from `RequestBuilder` the `openAPIClient` parameter can be omitted in the method signature.

## Describe alternatives you've considered

Same solution as in Alamofire template to pass interceptor to every request individually. Would probably be significantly more implementation effort

**Comment by Jonas1893** [comment]
I can also try and make a contribution for it if this is accepted

**Comment by wing328** [comment]
@Jonas1893 if it's not a lot of work, can you please file a PR to start with?

we definitely welcome contributions to enhance the generators

**Comment by Jonas1893** [comment]
> @Jonas1893 if it's not a lot of work, can you please file a PR to start with?
> 
> we definitely welcome contributions to enhance the generators

Here you go -> https://github.com/OpenAPITools/openapi-generator/pull/19904

**Comment by Jonas1893** [comment]
Resolved by #19904
